code/game/objects/items/devices/taperecorder.dm

```diff
@@ -164,69 +192,66 @@
 		return
 
 	if(recording)
-		recording = FALSE
-		mytape.timestamp += mytape.used_capacity
-		mytape.storedinfo += "\[[time2text(mytape.used_capacity * 10,"mm:ss")]\] Recording stopped."
 		playsound(src, 'sound/items/taperecorder/taperecorder_stop.ogg', 50, FALSE)
-		to_chat(usr, "<span class='notice'>Recording stopped.</span>")
-		return
+		say("Recording stopped.")
+		recording = FALSE
 	else if(playing)
-		playing = FALSE
-		var/turf/T = get_turf(src)
 		playsound(src, 'sound/items/taperecorder/taperecorder_stop.ogg', 50, FALSE)
-		T.visible_message("<font color=Maroon><B>Tape Recorder</B>: Playback stopped.</font>")
+		say("Playback stopped.")
+		playing = FALSE
+	time_warned = FALSE
 	update_icon()
-
+	update_sound()
 
 /obj/item/taperecorder/verb/play()
 	set name = "Play Tape"
 	set category = "Object"
 
 	if(!can_use(usr))
 		return
-	if(!mytape || mytape.ruined)
+	if(!mytape || mytape.unspooled)
 		return
 	if(recording)
 		return
 	if(playing)
 		return
 
-	playing = 1
+	playing = TRUE
 	update_icon()
-	to_chat(usr, "<span class='notice'>Playing started.</span>")
+	update_sound()
+	say("Playback started.")
 	playsound(src, 'sound/items/taperecorder/taperecorder_play.ogg', 50, FALSE)
 	var/used = mytape.used_capacity	//to stop runtimes when you eject the tape
 	var/max = mytape.max_capacity
-	for(var/i = 1, used <= max, sleep(10 * playsleepseconds))
+	for(var/i = 1, used <= max, sleep(playsleepseconds))
 		if(!mytape)
 			break
 		if(playing == FALSE)
 			break
 		if(mytape.storedinfo.len < i)
+			say("End of recording.")
 			break
-		say(mytape.storedinfo[i])
+		say("[mytape.storedinfo[i]]")
 		if(mytape.storedinfo.len < i + 1)
 			playsleepseconds = 1
-			sleep(10)
-			say("End of recording.")
+			sleep(1 SECONDS)
 		else
 			playsleepseconds = mytape.timestamp[i + 1] - mytape.timestamp[i]
-		if(playsleepseconds > 14)
-			sleep(10)
-			say("Skipping [playsleepseconds] seconds of silence")
-			playsleepseconds = 1
+		if(playsleepseconds > 14 SECONDS)
+			sleep(1 SECONDS)
+			say("Skipping [playsleepseconds] seconds of silence.")
+			playsleepseconds = 1 SECONDS
 		i++
```

Comment:
Maybe this too? I understand if it's a bit out of scope, but moving sleep loops to process is something we should do at some point